sample_metadata/parser/generic_parser.py

```diff
@@ -26,30 +42,73 @@
     AnalysisStatus,
 )
 
+
+# https://mypy.readthedocs.io/en/stable/runtime_troubles.html#using-new-additions-to-the-typing-module
+if sys.version_info >= (3, 8):
+    from typing import Literal
```

Comment:
Nice, didn't know about `Literal`! Looks simpler than using an Enum

Reply:
Brand new for me too!